node/lib/util/synthetic_gc_util.js

```
@@ -167,17 +172,26 @@ SyntheticGcUtil.prototype.getBareSubmoduleRepo = co.wrap(
         }
 
         if (!fs.existsSync(subPath)) {
-            console.log("Submodule at following path does not exists: " +
-                subPath);
+
+            if (this.d_verbose) {
+                console.log("Submodule at following path does not exist: " +
+                    subPath);
+            }
 
             const metaUrl = yield GitUtil.getOriginUrl(repo);
-            if (metaUrl !== null) {
-                subUrl = yield fetcher.getSubmoduleUrl(subName);
-                subUrl = SubmoduleConfigUtil.resolveSubmoduleUrl(metaUrl,
-                                                                 subUrl);
+            if (metaUrl === null) {
+                console.error("Cannot determine origin url. Something is " +
+                    "wrong, exiting.");
+                process.exit(-1);
```

should we throw an exception?